app/src/main/java/com/example/cooktaildb/base/BaseActivity.kt

```diff
@@ -0,0 +1,25 @@
+package com.example.cooktaildb.base
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.widget.ProgressBar
+import androidx.appcompat.app.AppCompatActivity
+import androidx.viewbinding.ViewBinding
+
+typealias ActivityInflate<T> = (LayoutInflater) -> T
+abstract class BaseActivity<B: ViewBinding>(val bindingFactory: ActivityInflate<B>) : AppCompatActivity() {
+    private var binding: B? = null
```

Comment:
```suggestion
    private var _binding: B? = null
    val binding get() = _binding
```

các accession chỉ được phép truy cập binding chứ ko đc modify , nên sẽ viết thế này
BaseFragment tương tự nha